### ISSUE TITLE
[consistent-type-imports] new rule request

### ISSUE THREAD

**Issue by villelahdenvuo** [angular-eslint/angular-eslint#1720]
**Description of the issue**

ESLint has a great rule https://typescript-eslint.io/rules/consistent-type-imports/ but it doesn't work with Angular as it doesn't detect the DI pattern of Angular and flags classes as being used as types when Angular needs to keep the references.

I was wondering if it would be too difficult to add the same rule to the Angular plugin extending the existing rule and just ignoring places where DI is being used e.g. constructors?

**Comment by reduckted** [comment]
I've had a look into whether this is possible. As a proof-of-concept, I copied the rule from `typescript-eslint` and adjusted it to consider types in constructors of injectable classes to be "values" instead of "types", which prevents the rule from saying they should be changed to type imports. 

For example, with this:
```ts
import { Service } from 'service';
@Component()
class Foo {
  constructor(service: Service) {}
                    // ~~~~~~~
}
```
The original rule considers the `Service` type annotation to be a "type reference". I changed the rule to consider it to _not_ be a "type reference", which prevents the rule from reporting a problem.

It wasn't too difficult, however it did require copying the entire rule, which is almost 1000 lines long.

I guess it's going to be a trade off between how valuable this rule would be, versus the maintenance burden of having to keep the rule's modified implementation in sync with the base rule's implementation.

@JamesHenry Do you have any thoughts about whether this rule is worth pursuing?

**Comment by villelahdenvuo** [comment]
I think that this rule would be quite beneficial because I have noticed that tree shaking sometimes does not remove everything correctly and switching to type imports can reduce the bundle size significantly in such scenarios.
